cli/azd/pkg/ext/event_dispatcher.go

```diff
@@ -41,6 +41,16 @@ func (ed *EventDispatcher[T]) AddHandler(name Event, handler EventHandlerFn[T])
 	}
 
 	events := ed.handlers[name]
+	newHandler := fmt.Sprintf("%v", handler)
+
+	// Ensure the handler is not already registered
+	for _, ref := range events {
+		existingHandler := fmt.Sprintf("%v", ref)
+		if existingHandler == newHandler {
+			return nil
+		}
+	}
```

Comment:
Ensures that an event handler is never incorrectly registered more than one time.  This can happen when multiple services of the same type exist within an azd project.

Reply:
Cool trick!  What does `fmt.Sprintf("%v", handler)` end up producing?  Is it some compiler generated name?

Reply:
I ended up rolling this back because it was not needed.. but basically just gets the pointer address as a string that could be used for comparisons.

However it didn't work exactly as intended since functions are passed by value - so even a function passed in from 2 different struct instances was returning the same value which was not intended.